src/Microsoft.VisualStudio.Composition.AppHost/CreateComposition.cs

```diff
@@ -44,6 +44,16 @@ public class CreateComposition : AppDomainIsolatedTask, ICancelableTask
         /// </summary>
         private readonly List<string> catalogAssemblyPaths = new List<string>();
 
+        /// <summary>
+        /// Gets the assemblies that have been logged as to their locations already.
+        /// </summary>
+        private readonly HashSet<Assembly> loggedAssemblies = new HashSet<Assembly>();
+
+        /// <summary>
+        /// Gets the names of assemblies that could not be found and have been logged.
+        /// </summary>
+        private readonly HashSet<string> loggedMissingAssemblyNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Why do we ignore case here, but not for loggedAssemblies above?

Reply:
Because we're storing strings in this set, whereas the *found* assemblies we're storing `Assembly` objects.